chatgpt/nodes/base_node.py

```diff
@@ -155,7 +155,7 @@ def sync(self):
         # Ensure validator hotkey is still registered on the network.
         self.check_registered()
 
-        if self.should_sync_state():
+        if current_block % self.config.node.epoch_length == 0:
```

Comment:
iirc, current_block isn't guaranteed to change every time this is called so resync_state could be called many times for the same block. I could be wrong, but maybe worth double-checking.

Reply:
That should be covered by the if statement on line 149.

From my testing, it's writing state to disk every 10 blocks. I also checked the created at + modified at dates and they match, meaning the file isn't being overwritten. 
<img width="815" alt="image" src="https://github.com/vana-com/vana-dlp-chatgpt/assets/5796181/b6d6f2d5-d340-467c-8e3e-e412940158a1">